src/main/java/org/zalando/nakadi/service/EventStreamWriterBinary.java

```diff
@@ -62,4 +82,59 @@ public int writeBatch(final OutputStream os, final Cursor cursor, final List<Str
         os.write(B_BATCH_SEPARATOR);
         return byteCount;
     }
+
+    @Override
+    public int writeSubscriptionBatch(final OutputStream os, final SubscriptionCursor cursor,
+                                      final List<ConsumedEvent> events,
+                                      final Optional<String> metadata) throws IOException {
+        int byteCount = B_FIXED_BYTE_COUNT_SUBSCRIPTION;
+
+        os.write(B_CURSOR_PARTITION_BEGIN);
+        final byte[] partition = cursor.getPartition().getBytes(StandardCharsets.UTF_8);
+        os.write(partition);
+        byteCount += partition.length;
+        os.write(B_OFFSET_BEGIN);
+        final byte[] offset = cursor.getOffset().getBytes(StandardCharsets.UTF_8);
+        os.write(offset);
+        byteCount += offset.length;
+
+        os.write(B_EVENT_TYPE_BEGIN);
+        final byte[] eventType = cursor.getEventType().getBytes(StandardCharsets.UTF_8);
+        os.write(eventType);
+        byteCount += eventType.length;
+
+        os.write(B_CURSOR_TOKEN_BEGIN);
+        final byte[] cursorToken = cursor.getCursorToken().getBytes(StandardCharsets.UTF_8);
+        os.write(cursorToken);
+        byteCount += cursorToken.length;
+
+        os.write(B_CURSOR_PARTITION_END);
+        if (!events.isEmpty()) {
+            os.write(B_EVENTS_ARRAY_BEGIN);
+            for (int i = 0; i < events.size(); i++) {
+                final byte[] event = events.get(i).getEvent().getBytes(StandardCharsets.UTF_8);
+                os.write(event);
+                byteCount += event.length;
+                if (i < (events.size() - 1)) {
+                    os.write(B_COMMA_DELIM);
+                } else {
+                    os.write(B_CLOSE_BRACKET);
+                }
+            }
+        }
+        if (metadata.isPresent()) {
+            os.write(B_DEBUG_BEGIN);
+            byteCount += B_DEBUG_BEGIN.length;
+
+            final byte[] debug = metadata.get().getBytes(StandardCharsets.UTF_8);
+            os.write(debug);
+            byteCount += debug.length;
```

Comment:
@antban I thought about it. Event type name is safe and I hope that debug will remain safe since it's generated by the Nakadi and is not result of user input.